runtime/battery-station/src/lib.rs

```diff
@@ -101,32 +112,88 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
     state_version: 1,
 };
 
+#[derive(scale_info::TypeInfo)]
+pub struct ContractsCallfilter;
+
+impl Contains<RuntimeCall> for ContractsCallfilter {
+    fn contains(runtime_call: &RuntimeCall) -> bool {
+        #[allow(clippy::match_like_matches_macro)]
+        match runtime_call {
+            RuntimeCall::AssetManager(transfer { .. }) => true,
+            RuntimeCall::PredictionMarkets(inner_call) => {
+                match inner_call {
+                    buy_complete_set { .. } => true,
+                    deploy_swap_pool_and_additional_liquidity { .. } => true,
+                    deploy_swap_pool_for_market { .. } => true,
+                    dispute { .. } => true,
+                    // Only allow CPMM markets using Authorized or SimpleDisputes dispute mechanism
+                    create_market {
+                        dispute_mechanism:
+                            MarketDisputeMechanism::Authorized | MarketDisputeMechanism::SimpleDisputes,
+                        scoring_rule: ScoringRule::CPMM,
+                        ..
+                    } => true,
+                    create_cpmm_market_and_deploy_assets {
+                        dispute_mechanism:
+                            MarketDisputeMechanism::Authorized | MarketDisputeMechanism::SimpleDisputes,
+                        ..
+                    } => true,
+                    edit_market {
+                        dispute_mechanism:
+                            MarketDisputeMechanism::Authorized | MarketDisputeMechanism::SimpleDisputes,
+                        scoring_rule: ScoringRule::CPMM,
+                        ..
+                    } => true,
+                    redeem_shares { .. } => true,
+                    report { .. } => true,
+                    sell_complete_set { .. } => true,
+                    start_global_dispute { .. } => true,
+                    _ => false,
+                }
+            }
+            RuntimeCall::Swaps(inner_call) => match inner_call {
+                pool_exit { .. } => true,
+                pool_exit_subsidy { .. } => true,
+                pool_exit_with_exact_asset_amount { .. } => true,
+                pool_exit_with_exact_pool_amount { .. } => true,
+                pool_join { .. } => true,
+                pool_join_with_exact_asset_amount { .. } => true,
+                pool_join_with_exact_pool_amount { .. } => true,
+                swap_exact_amount_in { .. } => true,
+                swap_exact_amount_out { .. } => true,
+                _ => false,
+            },
+            _ => false,
+        }
+    }
+}
+
 #[derive(scale_info::TypeInfo)]
 pub struct IsCallable;
 
 // Currently disables Court, Rikiddo and creation of markets using Court.
 impl Contains<RuntimeCall> for IsCallable {
     fn contains(call: &RuntimeCall) -> bool {
-        use zeitgeist_primitives::types::{
-            MarketDisputeMechanism::Court, ScoringRule::RikiddoSigmoidFeeMarketEma,
-        };
```

Comment:
Ok, these imports are not needed, because of [this](https://github.com/zeitgeistpm/zeitgeist/blob/7e82892d656ea1f0e56205b84c02662a4ef09a24/runtime/battery-station/src/lib.rs#L59).